File: 프로그래머스/0/181851. 전국 대회 선발 고사/전국 대회 선발 고사.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <utility>
using namespace std;

int solution(vector<int> rank, vector<bool> attendance) {
    int answer = 0;
    vector<pair<int,int>> result;
    for(int i = 0; i < attendance.size(); i++)
    {
        if(attendance[i])
            result.push_back({rank[i], i});
    }
    sort(result.begin(), result.end());
    answer = 10000 * result[0].second + 100 * result[1].second + result[2].second;
    return answer;
}

File: 프로그래머스/0/181851. 전국 대회 선발 고사/전국 대회 선발 고사.cs
using System;
using System.Collections.Generic;
public class Solution {
    public int solution(int[] rank, bool[] attendance) {
        int answer = 0;
        List<KeyValuePair<int, int>> result = new List<KeyValuePair<int, int>>();
        for(int i = 0; i < attendance.Length; i++)
        {
            if(attendance[i])
                result.Add(new KeyValuePair<int, int>(rank[i], i));
        }
        result.Sort((a, b) => a.Key.CompareTo(b.Key));
        answer = 10000 * result[0].Value + 100 * result[1].Value + result[2].Value;
        return answer;
    }
}